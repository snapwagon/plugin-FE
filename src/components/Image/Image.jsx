import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const Image = (props) => (
  <div
    aria-label={props.alt}
    className="coup-Thumbnail"
    style={{
      backgroundImage: `url(${props.src})`
    }}
    role="img"
  />
);

const {
  bool,
  string,
  objectOf,
  oneOfType
} = PropTypes;

Image.propTypes = {
  alt: string.isRequired,
  classNames: oneOfType([
    string,
    objectOf(bool)
  ]),
  src: string.isRequired,
  width: string
};

Image.defaultProps = {
  classNames: undefined,
  width: '100%'
};

export default Image;
