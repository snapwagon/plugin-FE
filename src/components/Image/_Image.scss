.snapW-Image {
  box-sizing: content-box;
}

.snapW-Thumbnail {
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  display: flex;
  flex: 1;
}
