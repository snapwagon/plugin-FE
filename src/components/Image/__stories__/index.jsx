/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Image from '../Image';

const mealImage = 'https://d2qs5c8ct3mfv5.cloudfront.net/images/standard_Business-4001-f7533f107822795023225e25b6f5dec8-Jumping.JPG';

storiesOf('Image', module)
  .addWithInfo('an <Image />', () => (
    <Image
      alt="Pork & Cabbage Tacos"
      src={mealImage}
    />
  ))
  .addWithInfo('an <Image /> in fixed-width parent', () => (
    <FixedWidth>
      <Image
        alt="Pork & Cabbage Tacos"
        src={mealImage}
      />
    </FixedWidth>
  ))
  .addWithInfo('an <Image /> with a custom classname', () => (
    <FixedWidth>
      <ScopedStyle rule=".my-Image { border: 4px solid magenta; }" />
      <Image
        alt="Pork & Cabbage Tacos"
        classNames="my-Image"
        src={mealImage}
      />
    </FixedWidth>
  ));

/* eslint-disable react/prop-types */

function FixedWidth({ children }) {
  return (
    <div
      style={{
        border: '2px solid #000',
        width: '300px',
        height: '350px',
        margin: '0 auto'
      }}
    >
      {children}
    </div>
  );
}

function ScopedStyle({ rule }) {
  return (
    <style type="text/css" scoped>
      {rule}
    </style>
  );
}
