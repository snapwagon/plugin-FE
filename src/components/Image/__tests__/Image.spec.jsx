/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';

import Image from '../Image.jsx';

describe('<Image /> snapshots', () => {
  test('rendered with required props', () => {
    const tree = renderer.create(
      <Image
        alt="Pork & Cabbage Tacos"
        src="/image.png"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('rendered with all props', () => {
    const tree = renderer.create(
      <Image
        alt="Pork & Cabbage Tacos"
        classNames={{
          'my-Image': true,
          'my-NonImage': false
        }}
        src="/image.png"
        width="80%"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
