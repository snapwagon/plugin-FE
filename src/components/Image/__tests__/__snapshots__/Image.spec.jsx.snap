// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> snapshots rendered with all props 1`] = `
<img
  alt="Pork & Cabbage Tacos"
  className="snapW-Image my-Image"
  src="/image.png"
  width="80%"
/>
`;

exports[`<Image /> snapshots rendered with required props 1`] = `
<img
  alt="Pork & Cabbage Tacos"
  className="snapW-Image"
  src="/image.png"
  width="100%"
/>
`;
